api/redisfailover/v1/types.go

```diff
@@ -43,6 +43,7 @@ type RedisSettings struct {
 	Port                          int32                             `json:"port,omitempty"`
 	Resources                     corev1.ResourceRequirements       `json:"resources,omitempty"`
 	CustomConfig                  []string                          `json:"customConfig,omitempty"`
+	MemoryOverheadPercentage      int32                             `json:"memoryOverheadPercentage,omitempty"`
```

Comment:
the naming is bit misleading , memory overhead means amount of memory thats overhead , but here i belive this number indicates percentage of memory that can be left off for redis to handle its other operations

better to name as AllowedmaxMemoryGrowthPercent and have it as 90